src/Symfony/Component/Cache/Tests/Adapter/MemcachedAdapterTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Tests\Adapter;
+
+use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedAdapterTest extends AdapterTestCase
+{
+    protected $skippedTests = array(
+        //'testExpiration' => 'Testing expiration slows down the test suite',
+        //'testHasItemReturnsFalseWhenDeferredItemIsExpired' => 'Testing expiration slows down the test suite',
+        //'testDefaultLifeTime' => 'Testing expiration slows down the test suite',
+    );
+
+    /**
+     * @var \Memcached
+     */
+    protected static $client;
+
+    protected static function defaultClientServer()
+    {
+        return sprintf(
+            'memcached://%s:%d',
+            getenv('MEMCACHED_HOST') ?: '127.0.0.1',
+            getenv('MEMCACHED_PORT') ?: 11211
+        );
+    }
+
+    private static function defaultClientOptions()
+    {
+        return array(
+            'OPT_LIBKETAMA_COMPATIBLE' => true,
+            'OPT_COMPRESSION' => true,
+        );
+    }
+
+    public static function setupBeforeClass()
+    {
+        if (!MemcachedAdapter::isSupported()) {
+            self::markTestSkipped('Extension memcached >2.1.0 required.');
+        }
+
+        $adapter = MemcachedAdapter::create(
+            static::defaultClientServer(),
+            static::defaultClientOptions()
+        );
+        static::$client = static::getClientFromAdapter($adapter);
+
+        parent::setupBeforeClass();
+    }
+
+    public static function tearDownAfterClass()
+    {
+        self::$client->flush();
```

Comment:
shouldn't it be flushed between tests instead of at the end of the testcase, to ensure test isolation ?

I would even go further and avoid making the client static, ensuring we don't reuse it between tests at all.